vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -207,8 +207,9 @@ type internal FSharpProjectOptionsManager
         match hier with
         | h when (h.IsCapabilityMatch("CPS")) ->
             let project = workspace.CurrentSolution.GetProject(projectId)
-            let siteProvider = provideProjectSiteProvider(workspace, project, serviceProvider, Some projectOptionsTable)
-            this.UpdateProjectInfo(tryGetOrCreateProjectId, projectId, siteProvider.GetProjectSite(), userOpName)
+            if project <> null then
```

Comment:
nit: is `x = null` or `x <> null` strictly less efficient than a pattern match or `isNull` call? Not sure if my knowledge here is rusty or not

Reply:
IIRC 'x = null' can throw NullReferenceException at runtime when x is null and the compiler thinks that it cannot be null.

Reply:
isNull x is probably the way to go.

Reply:
Hmmm!!! isNotNull is internal:  https://github.com/Microsoft/visualfsharp/blob/e6d46b9f56295aa70889ebff59cc0c6f0144a3ce/src/fsharp/FSharp.Core/prim-types.fs#L3324

I wonder why that is?

@dsyme any ideas?

Reply:
Because it changed fsharp.core surface area when I added it. And my changes where rolled back.

Reply:
In other words: it was public for a week or so

Reply:
See https://github.com/fsharp/fslang-suggestions/issues/99#issuecomment-266719119

> ... we have a design rule not to put "NotXYZ" predicates in the core library

Reply:
@dsyme ahh well this doen't fall under that rule because it is:  
   isNotNull     :-) 

It is a decent api we should make it public for FSharp.Core   4.3

Reply:
The design rule is clear both for F# and for .NET - we don't put negative "not" predicates in our libraries.  It is a slippery slope and we should not go down it. Just use ``not (isNull x)``.  

If that's not satisfactory we would be much better off putting effort into https://github.com/fsharp/fslang-suggestions/issues/569 to eventually allow ``!isNull x``